app/util/statusPoll.js

```diff
@@ -0,0 +1,98 @@
+const serviceState = {
+  UP: Symbol('status-up'),
+  DOWN: Symbol('status-down'),
+  ERROR: Symbol('status-error'),
+}
+const UNKNOWN = Symbol('status-unknown')
+const stateSymbols = new Set(Object.values(serviceState))
+
+const startStatusHandler = async ({ pollingPeriodMs, serviceTimeoutMs, getStatus }) => {
+  const status = {
+    status: UNKNOWN,
+    detail: null,
+  }
+  let timeoutHandle = null,
+    stop = false
+
+  const resetStatus = () => {
+    status.status = serviceState.ERROR
+    status.detail = null
+  }
+
+  const updateStatus = async () => {
+    try {
+      const newStatus = await Promise.race([
+        getStatus(),
+        new Promise((resolve) => setTimeout(resolve, serviceTimeoutMs, { status: serviceState.ERROR, detail: null })),
+      ])
+
+      if (stateSymbols.has(newStatus.status)) {
+        status.status = newStatus.status
+        status.detail = newStatus.detail === undefined ? null : newStatus.detail
+      } else {
+        resetStatus()
+      }
+    } catch (err) {
+      resetStatus()
+    }
+  }
+
+  const statusLoop = async () => {
+    await updateStatus()
+    if (!stop) {
+      timeoutHandle = setTimeout(statusLoop, pollingPeriodMs)
+    }
+  }
+
+  await statusLoop()
+
+  return {
+    get status() {
+      return status.status
+    },
+    get detail() {
+      return status.detail
+    },
+    close: () => {
+      stop = true
+      clearTimeout(timeoutHandle)
+    },
+  }
+}
+
+const buildCombinedHandler = async (handlerMap) => {
+  const getStatus = () =>
+    [...handlerMap].reduce((acc, [, h]) => {
```

Comment:
maybe call acc (accumulator) a `service`?

Reply:
I can rename `acc`. Probably to `accStatus` to make it clear it's a status

Reply:
done